org.metaborg.spoofax.meta.core/src/main/java/org/metaborg/spoofax/meta/core/pluto/build/main/GenerateSourcesBuilder.java

```diff
@@ -530,4 +544,45 @@ public static Origin origin(Input input) {
 
         return None.val;
     }
+
+    private void strategoIncrCompTasks(GenerateSourcesBuilder.Input input, File cacheDir, Arguments extraArgs,
+        File outputPath, Origin origin) throws IOException {
+        // TODO: use mainFile once we track imports
+        @SuppressWarnings("unused") File mainFile = input.strFile;
+
+        // FRONTEND
+        List<File> boilerplateFiles = new ArrayList<>();
+        Origin.Builder allFrontEndTasks = Origin.Builder();
+        BinaryRelation.Transient<String, File> generatedFiles = BinaryRelation.Transient.of();
+        Map<String, Origin.Builder> strategyOrigins = new HashMap<>();
+        for(File strFile : FileUtils.listFiles(context.baseDir, new String[] { "str" }, true)) {
```

Comment:
Looking for all `.str` files here does not take into account any included files and directories. You should take `List<File> strjIncludeDirs` and `List<File> strjIncludeFiles` from `GenerateSourcesBuilder.Input` into account.

Reply:
Additionally, when taking include dirs and files into account, there will probably be overlap in module names and/or paths to Stratego files, which may mess up your build script. Be sure to consider this in your binary relations and `persistentPath` implementations.